src/main/java/app/sami/languageWeb/request/RequestController.java

```diff
@@ -45,9 +46,51 @@ public RequestDto editRequest(@PathVariable UUID userId,
                                   @RequestJwtSubject UUID subject,
                                   @RequestBody EditRequestDto editRequestDto)
     {
-        if (subject != userId){
+        if (!subject.equals(userId)){
             throw new UserNotAllowedException();
         }
         return RequestMapper.toRequestDto(requestService.editRequest(editRequestDto, requestId));
     }
+
+    @GetMapping("/public/requests/summary")
+     public RequestLanguagesStatsDto getRequestLanguageStats(){
+        List<RequestLanguageStatsDto> requestLanguageStatsDtos = requestRepository.findNbRequestsPerLanguages()
+                .stream().map(RequestMapper::toRequestLanguageStats).toList();
+
+        return new RequestLanguagesStatsDto(requestLanguageStatsDtos);
+    }
+
+    @GetMapping("/requests/{fileName}/upload-uri")
+    public RequestUriDto getUploadUri(@PathVariable String fileName,
+                                      @RequestJwtSubject UUID subject){
+        return RequestUriDto.builder()
+                .url(requestService.getUploadUri(subject, fileName))
+                .fileName(fileName)
+                .build();
+    }
+    @GetMapping("/requests/{fileName}/download-uri")
```

Comment:
same here `/requests/download-uri?fileName=something`, but I'm not sure why the download URI is not generated with the `GET requests`